File: Python3/804. Unique Morse Code Words.py
class Solution:
    def uniqueMorseRepresentations(self, words):
        """
        :type words: List[str]
        :rtype: int
        """
        MORSE = [".-","-...","-.-.","-..",".","..-.","--.",
            "....","..",".---","-.-",".-..","--","-.","---",
            ".--.","--.-",".-.","...","-","..-","...-",".--",
            "-..-","-.--","--.."]
        transformations = []
        for word in words:
            transformation = ''
            for char in word:
                transformation += MORSE[ord(char)-97]
            transformations.append(transformation)
        numDict = collections.Counter(transformations)
        return len(numDict.keys())

File: Golang/804. Unique Morse Code Words.go
func uniqueMorseRepresentations(words []string) int {
    MORSE := [26]string{".-","-...","-.-.","-..",".","..-.","--.",
        "....","..",".---","-.-",".-..","--","-.","---",
        ".--.","--.-",".-.","...","-","..-","...-",".--",
        "-..-","-.--","--.."}
    transforDict := make(map[string]int)
    for _, word := range words {
        transfor := ""
        for _, char := range word {
            transfor += MORSE[int(char)-'a']
        }
        transforDict[transfor]++
    }
    return len(transforDict)
}
